src/plugins/intel_cpu/src/nodes/interpolate.cpp

```diff
@@ -272,7 +277,107 @@ struct jit_uni_interpolate_kernel_f32 : public jit_uni_interpolate_kernel, publi
             {}, {load_pool_gpr_idxs});
     }
 
+    /**
+     * @brief Auxilary worker function for emulate_gather_byte()
+     *
+     * @param xmm_arg Destination vector register
+     * @param mem_base Register with the base address
+     * @param xmm_index Vector register with indices
+     * @param mem_aux Auxilary 64-bit register
+     * @param byte_offset Offset of the current byte [0..3]
+     * @return none.
+     */
+    inline void emulate_gather_byte_impl(const Xbyak::Xmm &xmm_arg, const Xbyak::Reg64 &mem_base, const Xmm &xmm_index,
+                                         const Xbyak::Reg64 &mem_aux, int byte_offset) {
+        constexpr int xmm_len = cpu_isa_traits<sse41>::vlen;
+        const size_t src_prc_size = jcp_.src_prc.size();
+        if (src_prc_size != 1) {
+            IE_THROW() << "The data type of size '" << src_prc_size << "' is not supported.";
+        }
+        constexpr int ind_block_size = xmm_len / sizeof(int32_t);
+        for (int i = 0; i < ind_block_size; ++i) {
+            vpextrd(Xbyak::Reg32(mem_aux.getIdx()), xmm_index, i);
+            add(mem_aux, mem_base);
+            vpinsrb(xmm_arg, xmm_arg, ptr[mem_aux], i + byte_offset * ind_block_size);
+        }
+    }
```

Comment:
Are you able to emulate with simple mov() based on top of stack pointed by rsp. _extrd() and _insrb() have relative larger and varied latency on different platforms. We have similar gather emulation implementation in interpolate cubic mode. You can ref here:https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/nodes/interpolate.cpp#L1286-L1309

Reply:
Use uni_xxx(such as uni_vpinsrb()) version as much as possible.

1. Run vpinsrb() on sse41 will have illegal instruction error.

2. Run pinsrb() on avx2/avx512 have performance penalty sometimes. 

use uni_xxx can help you handle this.

Reply:
@chenhu-wang, Changed gather as you suggested, it gained some performance.